hikari/impl/entity_factory.py

```diff
@@ -424,10 +421,7 @@ def deserialize_audit_log(self, payload: data_binding.JSONObject) -> audit_log_m
                 user_id = snowflakes.Snowflake(raw_user_id)
 
             action_type: typing.Union[audit_log_models.AuditLogEventType, int]
-            try:
-                action_type = audit_log_models.AuditLogEventType(entry_payload["action_type"])
-            except ValueError:
-                action_type = entry_payload["action_type"]
+            action_type = audit_log_models.AuditLogEventType(entry_payload["action_type"])
```

Comment:
For the relevant audit log fields we were likely typing them as `Union[Enum, int]` on the model, I don't see that being removed anywhere and unless I missed it so would the Union with int now be unnecessary?

Reply:
No. The enum implementation handles it in a way that if we cast a value that is not found in the enum, it returns the raw value:
https://github.com/nekokatt/hikari/blob/1407e6c7565dd97c78b61e5cbbac41e7741cc8fd/hikari/utilities/enums.py#L117-L122

This is done so that enums dont break if Discord decides to suddenly change something

Reply:
So we need those unions everywhere then?

Reply:
We have to since we cant guarantee the user that the value will be in the enum if they use a older version of the lib that doesn't include something that Discord added. This just helps bots from randomly breaking due to Discord changing stuff

Reply:
But we do want our models to be typed accurately don't we? Seems weird to leave those unions in 2 arbitrary places if the behaviour is now present across all our models

Reply:
Oh, thought all the other ones already had that, will fix :+1: